### PR TITLE
Main thread setup.js file

### PR DESCRIPTION
**Pull request by greggraf**
Here is the diff for the next tweak.
- creates a new setup.js for the main thread
- moves directory creation out of client/lib/setup.js into the new setup.js
- general cleanup of some unused requires across the code base
- moved the datesource key generation logic out of app/client/models/datasources.js and into the new setup.js with IPC to share them back and forth

### CODE REVIEW

File: app/client/models/datasources.js [hunk 1]
```diff
@@ -40,10 +34,15 @@ module.exports = (state, bus) => {
...
+  const setlist = _list => {
+    state.datasources.list = _list
+  }
 
   const shown = () => state.datasources.shown
-  const setshown = _shown => state.datasources.shown = _shown
+
+  const setshown = _shown => {
+    state.datasources.shown = _shown
+  }
```
Comment (blahah) [inline app/client/models/datasources.js]: does the multiline format provide any benefit here?

File: app/client/lib/setup.js [hunk 2]
```diff
@@ -9,15 +9,9 @@ if (process.env.SCIENCEFAIR_DEVMODE) {
...
-
-mkdirp(C.DATAROOT)
-mkdirp(C.COLLECTION_PATH)
-mkdirp(C.DATASOURCES_PATH)
 
 const datasources = require('./getdatasource')
 
 require('electron').ipcRenderer.on('quitting', () => {
-  datasources.all().forEach(d => d.close())
+  datasources.all().forEach(d => d.close(() => {}))
```
Comment (blahah) [inline app/client/lib/setup.js]: 👍

File: app/constants.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 const path = require('path')
 
-const { app } = require('electron').remote
+const app = require('electron').app
+const remote = require('electron').remote
+const api = app || remote.app
```
Comment (blahah) [inline app/constants.js]: 👍

Comment (greggraf) [inline app/constants.js]: I was noticing the unused `requires` myself, then to be sure I didn't miss any, I ran `eshint` (with `standard` as the preset) on just `datasources.js` and I was seeing `Arrow function should not return assignment` errors, so I tweaked those multiline functions.

I could take a side trip and add `eshint` to the project and see how much has to change, with the goal of making it automated. 

My next "i could add" would be about unit testing...

so I updated those.

Comment (blahah) [inline app/constants.js]: @greggraf I think the simplest linting add would be `npm install --save-dev standard`, then `"test": "standard"` in the scripts section of `package.json`. Eslint is overkill for now probably.

Unit testing should wait until we have the units figured out - which is essentially this series of PRs plus some modularisation. My preference would be to add tests to each module as we break it out into a separate repo.

Comment (blahah) [inline app/constants.js]: also, LGTM :shipit:

**Comment by blahah** [comment]
I forgot to say: thanks! I really appreciate your engagement and you've been moving this forward carefully but efficiently which is making for a really smooth transition :)

**Comment by greggraf** [comment]
thumbs up

**Comment by greggraf** [comment]
@blahah looking into an error I am seeing on app startup now...

**Pull request merged**